### PR TITLE
rec: Don't negcache entries for longer than their RRSIG validity

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/test-syncres_cc.cc [hunk 1]
```diff
@@ -278,10 +278,10 @@ static bool addRRSIG(const testkeysset_t& keys, std::vector<DNSRecord>& records,
   }
 
   DNSRecord rec;
+  rec.d_type = QType::RRSIG;
   rec.d_place = records[recordsCount-1].d_place;
   rec.d_name = records[recordsCount-1].d_name;
-  rec.d_type = QType::RRSIG;
-  rec.d_ttl = sigValidity;
+  rec.d_ttl = records[recordsCount-1].d_ttl;
```
Comment (Habbie) [inline pdns/recursordist/test-syncres_cc.cc]: shouldn't we cap this to the max of `d_ttl` and `sigValidity`?

Comment (rgacogne) [reply pdns/recursordist/test-syncres_cc.cc]: Section 3 of rfc4034 [1] states:

> The TTL value of an RRSIG RR MUST match the TTL value of the RRset it covers.

so this looks right to me?

[1]: https://tools.ietf.org/html/rfc4034#section-3

Comment (Habbie) [reply pdns/recursordist/test-syncres_cc.cc]: I may be confused, but say we receive an RRSIG with an 86400 TTL but expiry is in a few hours. We should only cache it for those few hours, and I get the impression this change would cache it for a day?

Comment (rgacogne) [reply pdns/recursordist/test-syncres_cc.cc]: Well this code is only there to generate mockup `RRSIG`s records for our unit tests, so it is intentionally designed to allow the creation of weird records :-)

Comment (Habbie) [reply pdns/recursordist/test-syncres_cc.cc]: durrrr

**Pull request merged**